### ISSUE TITLE
[endpoint] Data for Account

### ISSUE THREAD

**Issue by kbacha** [kbacha/stellar-sdk#50]
This endpoint represents a single data associated with a given account.

```rust
use stellar_client::endpoint::account::Data
```

https://www.stellar.org/developers/horizon/reference/endpoints/data-for-account.html

**Comment by robertDurst** [comment]
Thoughts on this, I could tackle this next?

**Comment by kbacha** [comment]
For sure. We don't actually have a resource representation for account data yet but I believe it's a hash map of sorts.

**Comment by robertDurst** [comment]
Awesome, fun!

**Comment by robertDurst** [comment]
Yup, looks like a hash map. Here it is from the Go SDK:

```
type Account struct {
	Links struct {
		Self         Link `json:"self"`
		Transactions Link `json:"transactions"`
		Operations   Link `json:"operations"`
		Payments     Link `json:"payments"`
		Effects      Link `json:"effects"`
		Offers       Link `json:"offers"`
	} `json:"_links"`

	HistoryAccount
	Sequence             string            `json:"sequence"`
	SubentryCount        int32             `json:"subentry_count"`
	InflationDestination string            `json:"inflation_destination,omitempty"`
	HomeDomain           string            `json:"home_domain,omitempty"`
	Thresholds           AccountThresholds `json:"thresholds"`
	Flags                AccountFlags      `json:"flags"`
	Balances             []Balance         `json:"balances"`
	Signers              []Signer          `json:"signers"`
	Data                 map[string]string `json:"data"`
}
```

**Comment by robertDurst** [comment]
So it looks like, while data is technically attached to an account, the response simply looks like this:

```
{
  "value": "UGl6emE="
}
```

As seen here: https://horizon-testnet.stellar.org/accounts/GATLAI2D7SSH6PE3HXTDPTRM4RE5VRK6HGA63K5EWP75PSANCZRFDNB5/data/Food

So what is the best way to go about this? Should I create a data resource that is simply something like this:

```
use deserialize;

#[derive(Serialize, Deserialize, Debug)]
pub struct Data {
    value: String,
}

impl Data {
    /// Key/value pairs associated with the account.
    pub fn value(&self) -> &String {
        &self.value
    }
}
```

So that in the actual endpoint I can do something like this:

```
impl EndPoint for Data {
    type Response = Data;

    fn into_request(self, host: &str) -> Result<Request<Body>> {
        let uri = Uri::from_str(&format!("{}/accounts/{}/data/{}", host, self.id, self.key))?;
        let request = Request::get(uri).body(Body::None)?;
        Ok(request)
    }
}
```

## **TLDR:** 'Do we need a separate resource just for this?

**Comment by kbacha** [comment]
Oh, so the endpoint is actually just a base64 encoded value. I think we should definitely model it but we may want to decode the base64 into the value as part of the deserialization. This crate looks pretty good:
https://crates.io/crates/base64

Of course, we also want this to be symmetric with the data response coming from the account resource. This makes me think that we actually want a data type called `Base64String` and then the data field on account is a hashmap of `HashMap<String, Base64String>`. and the `Data` response for the data endpoint is what you have above but the value is also a `Base64String`

```rust
#[derive(Serialize, Deserialize, Debug)]
pub struct Data {
    value: Base64String,
}
```

The key part here is creating the `Base64String` struct (similar to our `Amount` type) and implementing deserialize on it.

**Comment by robertDurst** [comment]
Hey so, I want to confirm I need to complete the following for this:```
1. Add account Data endpoint
2. Create Data resource
3. Create Base64String type (just like amount) and implement deserialize it
4. Add Data field to the Account struct in account resource
```

**Comment by kbacha** [comment]
@robertDurst Yup, but not necessarily in that order.

**Comment by robertDurst** [comment]
@kbacha Should Base64String store a plain utf8 String, serialize to Base64, and deserialize to plain utf8 String? 

This is what I have. Just wanted to check, is this backwards?
